Source/WebKit/UIProcess/WebAuthentication/Cocoa/WebAuthenticatorCoordinatorProxy.mm

```diff
@@ -453,6 +453,16 @@ static inline bool isCrossPlatformRequest(const Vector<AuthenticatorTransport>&
     }
 }
 
+inline static Vector<AuthenticatorTransport> toTransports(NSArray<ASAuthorizationSecurityKeyPublicKeyCredentialDescriptorTransport> *asTransports)
+{
+    Vector<AuthenticatorTransport> transports;
+    for (ASAuthorizationSecurityKeyPublicKeyCredentialDescriptorTransport asTransport : asTransports) {
+        if (auto transport = convertStringToAuthenticatorTransport(asTransport))
+            transports.append(*transport);
+    }
+    return transports;
+}
+
```

Comment:
In fact, you might be able to just use the `wtf/VectorCocoa.h` implementation of `toVector(NSArray)` by writing an appropriate `makeVectorElement` function:

```
//    std::optional<VectorElementType> makeVectorElement(const VectorElementType*, id arrayElement);
std::optional<AuthenticatorTransport> makeVectorElement(const AuthenticatorTransport*, ASAuthorizationSecurityKeyPublicKeyCredentialDescriptorTransport *arrayElement) {
    return convertStringToAuthenticatorTransport(asTransport));
}

Then below you can just say:

`response.transports = toVector(credential.get().transports);`

And get rid of the whole `toTransports()` method.

Reply:
(You probably need to add an include for `wtf/VectorCocoa.h`)